diff -Nurp automake-1.16.3.orig/lib/config.guess automake-1.16.3/lib/config.guess
--- automake-1.16.3.orig/lib/config.guess	2020-11-16 05:39:53.000000000 +0000
+++ automake-1.16.3/lib/config.guess	2021-05-17 12:48:39.782829260 +0000
@@ -1040,6 +1040,9 @@ EOF
     mips64el:Linux:*:*)
 	echo "$UNAME_MACHINE"-unknown-linux-"$LIBC"
 	exit ;;
+    loongarch64:Linux:*:* | loongarch32:Linux:*:*)
+	echo "$UNAME_MACHINE"-unknown-linux-"$LIBC"
+	exit ;;
     openrisc*:Linux:*:*)
 	echo or1k-unknown-linux-"$LIBC"
 	exit ;;
diff -Nurp automake-1.16.3.orig/lib/config.sub automake-1.16.3/lib/config.sub
--- automake-1.16.3.orig/lib/config.sub	2020-11-16 05:39:54.000000000 +0000
+++ automake-1.16.3/lib/config.sub	2021-05-17 12:48:04.342621470 +0000
@@ -1190,6 +1190,7 @@ case $cpu-$vendor in
 			| m6811 | m68hc11 | m6812 | m68hc12 | m68hcs12x \
 			| m88110 | m88k | maxq | mb | mcore | mep | metag \
 			| microblaze | microblazeel \
+			| loongarch64 | loongarch32 \
 			| mips | mipsbe | mipseb | mipsel | mipsle \
 			| mips16 \
 			| mips64 | mips64eb | mips64el \
