diff -Nurp cmake-3.22.3.orig/Modules/CheckTypeSize.c.in cmake-3.22.3/Modules/CheckTypeSize.c.in
--- cmake-3.22.3.orig/Modules/CheckTypeSize.c.in	2022-03-04 17:06:35.000000000 +0000
+++ cmake-3.22.3/Modules/CheckTypeSize.c.in	2022-03-18 08:17:38.518281758 +0000
@@ -13,6 +13,8 @@
 # define KEY '_','_','P','P','C','_','_'
 #elif defined(__ppc__)
 # define KEY '_','_','p','p','c','_','_'
+#elif defined(__loongarch_lp64)
+# define KEY '_','_','l','o','o','n','g','a','r','c','h','_','l','p','6','4'
 #elif defined(__aarch64__)
 # define KEY '_','_','a','a','r','c','h','6','4','_','_'
 #elif defined(__ARM_ARCH_7A__)
