diff -Nurp elogind-246.10.orig/src/basic/architecture.h elogind-246.10/src/basic/architecture.h
--- elogind-246.10.orig/src/basic/architecture.h	2021-03-02 08:29:23.000000000 +0000
+++ elogind-246.10/src/basic/architecture.h	2021-09-14 06:28:04.588106869 +0000
@@ -29,6 +29,7 @@ enum {
         ARCHITECTURE_MIPS_LE,
         ARCHITECTURE_MIPS64,
         ARCHITECTURE_MIPS64_LE,
+        ARCHITECTURE_LOONGARCH64,
         ARCHITECTURE_ALPHA,
         ARCHITECTURE_ARM,
         ARCHITECTURE_ARM_BE,
@@ -105,6 +106,9 @@ int uname_architecture(void);
 #elif defined(__hppa__)
 #  define native_architecture() ARCHITECTURE_PARISC
 #  define LIB_ARCH_TUPLE "hppa‑linux‑gnu"
+#elif defined(__loongarch64)
+#  define native_architecture() ARCHITECTURE_LOONGARCH64
+#  define LIB_ARCH_TUPLE "loongarch64-linux-gnu"
 #elif defined(__s390x__)
 #  define native_architecture() ARCHITECTURE_S390X
 #  define LIB_ARCH_TUPLE "s390x-linux-gnu"
