diff -Nurp gc-8.0.6.orig/include/private/gcconfig.h gc-8.0.6/include/private/gcconfig.h
--- gc-8.0.6.orig/include/private/gcconfig.h	2021-09-28 20:47:22.000000000 +0000
+++ gc-8.0.6/include/private/gcconfig.h	2021-12-16 08:05:35.137556347 +0000
@@ -238,6 +238,10 @@ EXTERN_C_BEGIN
 #    define VAX
 #    define mach_type_known
 # endif
+# if defined(__loongarch_lp64) 
+#    define LOONGARCH64
+#    define mach_type_known
+# endif
 # if defined(mips) || defined(__mips) || defined(_mips)
 #    define MIPS
 #    if defined(nec_ews) || defined(_nec_ews)
@@ -1732,6 +1736,26 @@ EXTERN_C_BEGIN
 #   define STACKBOTTOM ((ptr_t)0xfffff000) /* for Encore */
 # endif
 
+# ifdef LOONGARCH64
+#   define MACH_TYPE "LOONGARCH64"
+#   ifdef LINUX
+#     define OS_TYPE "LINUX"
+#     define DYNAMIC_LOADING
+      extern int _end[];
+#     pragma weak __data_start
+      extern int __data_start[];
+#     define DATASTART ((ptr_t)(__data_start))
+#     define DATAEND ((ptr_t)(_end))
+#     define CPP_WORDSZ 64
+#     define ALIGNMENT (64/8)
+#     ifndef HBLKSIZE
+#       define HBLKSIZE 4096
+#     endif
+#     define LINUX_STACKBOTTOM
+#   endif
+# endif
+
+
 # ifdef MIPS
 #   define MACH_TYPE "MIPS"
 #   ifdef LINUX
diff -Nurp gc-8.0.6.orig/os_dep.c gc-8.0.6/os_dep.c
--- gc-8.0.6.orig/os_dep.c	2021-09-28 20:47:22.000000000 +0000
+++ gc-8.0.6/os_dep.c	2021-12-16 08:06:36.491553769 +0000
@@ -3144,7 +3144,7 @@ GC_API GC_push_other_roots_proc GC_CALL
 #     ifndef SEGV_ACCERR
 #       define SEGV_ACCERR 2
 #     endif
-#     if defined(AARCH64) || defined(ARM32) || defined(MIPS) \
+#     if defined(AARCH64) || defined(ARM32) || defined(MIPS) || defined(LOONGARCH64) \
          || __FreeBSD__ >= 7
 #       define CODE_OK (si -> si_code == SEGV_ACCERR)
 #     elif defined(POWERPC)
