--- gcc-12.0.0.orig/gcc/config/loongarch/linux.h	2021-08-13 09:46:28.000000000 +0000
+++ gcc-12.0.0/gcc/config/loongarch/linux.h	2021-08-26 09:38:27.590357817 +0000
@@ -31,8 +31,8 @@ along with GCC; see the file COPYING3.
 #define GNU_USER_LINK_EMULATION32 "elf32loongarch"
 #define GNU_USER_LINK_EMULATION64 "elf64loongarch"
 
-#define GLIBC_DYNAMIC_LINKERLP32 "/lib32/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERLP64 "/lib64/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERLP32 "/lib32/ld-linux-loongarch32.so.1"
+#define GLIBC_DYNAMIC_LINKERLP64 "/lib64/ld-linux-loongarch64.so.1"
 
 #define GNU_USER_DYNAMIC_LINKERLP32 GLIBC_DYNAMIC_LINKERLP32
 #define GNU_USER_DYNAMIC_LINKERLP64 GLIBC_DYNAMIC_LINKERLP64
