diff -Nurp gcc-8.3.0.orig/gcc/config/loongarch/loongarch.h gcc-8.3.0/gcc/config/loongarch/loongarch.h
--- gcc-8.3.0.orig/gcc/config/loongarch/loongarch.h	2021-05-13 16:34:46.934439782 +0800
+++ gcc-8.3.0/gcc/config/loongarch/loongarch.h	2021-05-13 16:37:35.108129050 +0800
@@ -2313,3 +2313,11 @@ struct GTY(())  machine_function {
 #define FRAME_GROWS_DOWNWARD 1
 
 #define FUNCTION_VALUE_REGNO_P(N) ((N) == GP_RETURN || (N) == FP_RETURN)
+
+#if LARCH_ABI_DEFAULT == ABILP32
+#define STANDARD_STARTFILE_PREFIX_1 "/lib/"
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/lib/"
+#elif LARCH_ABI_DEFAULT == ABILP64
+#define STANDARD_STARTFILE_PREFIX_1 "/lib64/"
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/lib64/"
+#endif
