diff -Nurp glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/ldd-rewrite.sed glibc-2.33/sysdeps/unix/sysv/linux/loongarch/ldd-rewrite.sed
--- glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/ldd-rewrite.sed	2021-08-07 03:03:17.000000000 +0000
+++ glibc-2.33/sysdeps/unix/sysv/linux/loongarch/ldd-rewrite.sed	2021-08-07 07:16:15.144122523 +0000
@@ -1 +1 @@
-s_^\(RTLDLIST=\)\(.*lib/\)\(ld-linux\)-\(loongarch64\)-\(lp64\)\(d*\)\(\.so\.[0-9.]*\)_\1"\2\3-\4-\5\7 \2\3-\4-\5d\7"_
+s_^\(RTLDLIST=\)\(.*lib\)\(\|64\)\(/[^/]*\)\(-loongarch\)\(64\|32\)\(\.so\.[0-9.]*\)_\1"\2\3\4\564\7 \2\4\532\7"_
diff -Nurp glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/lp64/shlib-versions glibc-2.33/sysdeps/unix/sysv/linux/loongarch/lp64/shlib-versions
--- glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/lp64/shlib-versions	1970-01-01 00:00:00.000000000 +0000
+++ glibc-2.33/sysdeps/unix/sysv/linux/loongarch/lp64/shlib-versions	2021-08-07 06:26:14.352940485 +0000
@@ -0,0 +1,3 @@
+DEFAULT		GLIBC_2.27
+
+ld=ld-linux-loongarch64.so.1
diff -Nurp glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/shlib-versions glibc-2.33/sysdeps/unix/sysv/linux/loongarch/shlib-versions
--- glibc-2.33.orig/sysdeps/unix/sysv/linux/loongarch/shlib-versions	2021-08-07 03:03:17.000000000 +0000
+++ glibc-2.33/sysdeps/unix/sysv/linux/loongarch/shlib-versions	1970-01-01 00:00:00.000000000 +0000
@@ -1 +0,0 @@
-DEFAULT		GLIBC_2.27
