--- core.orig/configure.ac	2022-08-12 01:32:28.467617736 +0000
+++ core/configure.ac	2022-08-12 09:19:47.358439625 +0000
@@ -5672,7 +5672,7 @@ if test "$cross_compiling" = "yes"; then
     unset AR LD NM OBJDUMP PKG_CONFIG RANLIB READELF STRIP
     unset CPPUNIT_CFLAGS CPPUNIT_LIBS
     unset LIBXML_CFLAGS LIBXML_LIBS LIBXSLT_CFLAGS LIBXSLT_LIBS XSLTPROC
-    unset PKG_CONFIG_LIBDIR PKG_CONFIG_PATH
+    unset PKG_CONFIG_LIBDIR PKG_CONFIG_PATH PKG_CONFIG_SYSROOT_DIR
     if test -n "$CC_FOR_BUILD"; then
         export CC="$CC_FOR_BUILD"
         CC_BASE=`first_arg_basename "$CC"`
--- core.orig/aclocal.m4	2022-08-12 09:04:51.910477249 +0000
+++ core/aclocal.m4	2022-08-12 09:21:23.798435573 +0000
@@ -322,7 +322,7 @@ AC_DEFUN([AM_PATH_PYTHON],
   dnl Find a Python interpreter.  Python versions prior to 2.0 are not
   dnl supported. (2.0 was released on October 16, 2000).
   m4_define_default([_AM_PYTHON_INTERPRETER_LIST],
-[python python2 python3 dnl
+[python3 python python2 dnl
  python3.9 python3.8 python3.7 python3.6 python3.5 python3.4 python3.3 dnl
  python3.2 python3.1 python3.0 dnl
  python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 dnl
