From 44aa123bd9751bba81bbdcdd40dcab57f17dd7e4 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 16 Mar 2026 09:54:37 +0000
Subject: [PATCH] remove vc_uniscr_check function for paste selection.

---
 drivers/tty/vt/vt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/tty/vt/vt.c b/drivers/tty/vt/vt.c
index dea1d0e3e..3ef6c4208 100644
--- a/drivers/tty/vt/vt.c
+++ b/drivers/tty/vt/vt.c
@@ -3033,7 +3033,7 @@ static int vc_process_ucs(struct vc_data *vc, int *c, int *tc)
 		 * required. This is one such case as we need to remember
 		 * which displayed characters are double-width.
 		 */
-		vc_uniscr_check(vc);
+//		vc_uniscr_check(vc);
 		return 2;
 	}
 
@@ -3109,7 +3109,7 @@ static int vc_get_glyph(struct vc_data *vc, int tc)
 	 * character meaning the simple screen buffer won't hold the original
 	 * information, whereas the Unicode screen buffer always does.
 	 */
-	vc_uniscr_check(vc);
+//	vc_uniscr_check(vc);
 
 	/* Try getting a simpler fallback character. */
 	tc = ucs_get_fallback(tc);
-- 
2.31.1

