From 6646b0071cc5d43d9ee7ce9a00a610b3f66e3e32 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Mon, 9 Oct 2023 11:33:27 +0000
Subject: [PATCH] jack2 fix for python3.12

---
 waflib/Context.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/waflib/Context.py b/waflib/Context.py
index 761b521..d4406d8 100644
--- a/waflib/Context.py
+++ b/waflib/Context.py
@@ -6,7 +6,7 @@
 Classes and functions enabling the command system
 """
 
-import os, re, imp, sys
+import os, re, types, sys
 from waflib import Utils, Errors, Logs
 import waflib.Node
 
@@ -660,7 +660,7 @@ def load_module(path, encoding=None):
 	except KeyError:
 		pass
 
-	module = imp.new_module(WSCRIPT_FILE)
+	module = types.ModuleType(WSCRIPT_FILE)
 	try:
 		code = Utils.readf(path, m='r', encoding=encoding)
 	except EnvironmentError:
-- 
2.31.1

