From feda1af52e12ff69dc8c5421ce06f37986cfc513 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Wed, 24 May 2023 14:52:05 +0000
Subject: [PATCH 2/2] Fix for gcc 13.

---
 libfrog/randbytes.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libfrog/randbytes.h b/libfrog/randbytes.h
index 00fd7c4..aa9e3cf 100644
--- a/libfrog/randbytes.h
+++ b/libfrog/randbytes.h
@@ -5,6 +5,7 @@
  */
 #ifndef __LIBFROG_RANDBYTES_H__
 #define __LIBFROG_RANDBYTES_H__
+#include <stdint.h>
 
 extern uint8_t randbytes_test_buf[];
 
-- 
2.31.1

