From f9bab5b6eba75166b907aa10170ee1fc8b2831bb Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Sat, 7 Oct 2023 01:26:51 +0000
Subject: [PATCH] libglvnd add loongarch asm for meson build.

---
 meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index 236c188..5658195 100644
--- a/meson.build
+++ b/meson.build
@@ -82,6 +82,8 @@ if not with_asm.disabled()
     add_project_arguments('-DUSE_AARCH64_ASM', language : 'c')
   elif host_machine.cpu_family() == 'ppc64' and cc.get_define('_CALL_ELF') == '2'
     add_project_arguments('-DUSE_PPC64_ASM', language : 'c')
+  elif host_machine.cpu_family() == 'loongarch64'
+    add_project_arguments('-DUSE_LOONGARCH64_ASM', language : 'c')
   elif with_asm.enabled()
     error('No ASM available for @0@ (@1@ endian)'.format(host_machine.system(), host_machine.endian()))
   else
@@ -161,6 +163,8 @@ if use_asm
     gl_dispatch_type = 'armv7_tsd'
   elif host_machine.cpu_family() == 'aarch64'
     gl_dispatch_type = 'aarch64_tsd'
+  elif host_machine.cpu_family() == 'loongarch64'
+    gl_dispatch_type = 'loongarch64_tsd'
   elif host_machine.cpu_family() == 'ppc64'
     gl_dispatch_type = 'ppc64_@0@'.format(thread_type)
   endif
-- 
2.31.1

