From 8f94dfcf3297098e1683caddce5dd27441007618 Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Tue, 1 Aug 2023 06:24:31 +0000
Subject: [PATCH] fix configure error under loongarch64 architecture.

---
 lib/libcrypt.minver | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libcrypt.minver b/lib/libcrypt.minver
index 8478077..55fd360 100644
--- a/lib/libcrypt.minver
+++ b/lib/libcrypt.minver
@@ -60,7 +60,7 @@ GLIBC_2.3    kfreebsd.*gnu   i[3-9]86
 ERROR        kfreebsd.*gnu   .
 
 # Linux with GNU libc
-GLIBC_2.36   linux.*gnu      loongarch64   __WORDSIZE == 64 && ULONG_MAX != UINT_MAX /* lp64* ABI */
+GLIBC_2.36   linux.*gnu      loongarch64   defined __loongarch_lp64 && ULONG_MAX != UINT_MAX /* lp64* ABI */
 GLIBC_2.35   linux.*gnu      or1k
 GLIBC_2.33   linux.*gnu      riscv32
 GLIBC_2.32   linux.*gnu      arc
-- 
2.31.1

